internal/testing/e2e/backfill_test.go

```diff
@@ -321,14 +302,19 @@ func TestBackfillGenerationMismatch(t *testing.T) {
 	require.Nil(t, err)
 
 	resp, err := stream.Recv()
+	require.Nil(t, err)
+	bfID := resp.Match.Backfill.Id
+
+	resp, err = stream.Recv()
 	require.Nil(t, resp)
 	require.Equal(t, io.EOF, err)
-}
 
-func mustAny(m proto.Message) *any.Any {
-	result, err := ptypes.MarshalAny(m)
-	if err != nil {
-		panic(err)
-	}
-	return result
+	actual, err := om.Frontend().GetBackfill(ctx, &pb.GetBackfillRequest{BackfillId: bfID})
```

Comment:
Why do you test GetBackfill here?
The name of the function is createMatchWithBackfill and you are passing backfill as a parameter.
I think if you want to test GetBackfill you should do it out of this method in order to not overload one with a lot of code. createMatchWithBackfill does more than a simple create match step here.

Reply:
This is a refactor of @yeukovichd test. Now it is used in 3 place, so that we can associate ticket with backfill using a match proposal.

Reply:
One more additional check is not bad. So I will not remove this line.